cupy/padding/pad.py

```diff
@@ -1,321 +1,742 @@
 import numpy
-import six
 
 import cupy
 
 
-def _prepend_const(narray, pad_amount, value, axis=-1):
-    if pad_amount == 0:
-        return narray
-    padshape = tuple([x if i != axis else pad_amount
-                      for i, x in enumerate(narray.shape)])
-    return cupy.concatenate((cupy.full(padshape, value, narray.dtype),
-                             narray), axis=axis)
+###############################################################################
+# Private utility functions.
 
 
-def _append_const(narray, pad_amount, value, axis=-1):
-    if pad_amount == 0:
-        return narray
-    padshape = tuple([x if i != axis else pad_amount
-                      for i, x in enumerate(narray.shape)])
-    return cupy.concatenate((narray,
-                             cupy.full(padshape, value, narray.dtype)),
-                            axis=axis)
+def _round_if_needed(arr, dtype):
+    """Rounds arr inplace if destination dtype is integer.
+    """
+    if cupy.issubdtype(dtype, cupy.integer):
+        arr.round(out=arr)  # bug in round so use rint (cupy/cupy#2330)
+
+
+def _slice_at_axis(sl, axis):
+    """Construct tuple of slices to slice an array in the given dimension.
```

Comment:
Please use third-party singular present form for function docstrings.
(ditto for all other functions)

Reply:
I tried to update these, but not sure if I always did it exactly how you wanted. Feel free to make additional edits by pushing to my branch if that is easier than commenting individually.

I guess `pad` itself is the only function in the module that shows up in the user documentation, so we should pay closest attention to that one. The rendering of that looks much better after the latest commit.